src/pages/sales-order/Tabs/SalesOrderForm.js

```diff
@@ -0,0 +1,1364 @@
+    }
+  ]
+
+  async function fillForm(soHeader, soItems) {
+    const shipAdd = await getAddress(soHeader?.record?.shipToAddressId)
+    const billAdd = await getAddress(soHeader?.record?.billToAddressId)
+
+    soHeader?.record?.tdType == 1 || soHeader?.record?.tdType == null
+      ? setCycleButtonState({ text: '123', value: 1 })
+      : setCycleButtonState({ text: '%', value: 2 })
+
+    const modifiedList = soItems.list?.map((item, index) => ({
+      ...item,
+      id: index + 1,
+      basePrice: parseFloat(item.basePrice).toFixed(5),
+      unitPrice: parseFloat(item.unitPrice).toFixed(3),
+      upo: parseFloat(item.upo).toFixed(2),
+      vatAmount: parseFloat(item.vatAmount).toFixed(2),
+      extendedPrice: parseFloat(item.extendedPrice).toFixed(2)
+    }))
+
+    formik.setValues({
+      ...soHeader.record,
+      tdAmount:
+        soHeader?.record?.tdType == 1 || soHeader?.record?.tdType == null
+          ? soHeader?.record?.tdAmount
+          : soHeader?.record?.tdPct,
+      shipAddress: shipAdd,
+      billAddress: billAdd,
+      currentDiscount: soHeader?.record?.tdAmount,
+      items: modifiedList
+    })
+  }
+
+  async function getSalesOrder(soId) {
+    try {
+      const res = await getRequest({
+        extension: SaleRepository.SalesOrder.get,
+        parameters: `_recordId=${soId}`
+      })
+
+      res.record.date = formatDateFromApi(res?.record?.date)
+
+      return res
+    } catch (error) {}
+  }
+
+  async function getSalesOrderItems(soId) {
+    try {
+      return await getRequest({
+        extension: SaleRepository.SalesOrderItem.qry,
+        parameters: `_params=1|${soId}`
+      })
+    } catch (error) {}
+  }
+
+  async function getAddress(addressId) {
+    if (!addressId) return null
+
+    try {
+      const res = await getRequest({
+        extension: SystemRepository.FormattedAddress.get,
+        parameters: `_addressId=${addressId}`
+      })
+
+      return res?.record?.formattedAddress.replace(/(\r\n|\r|\n)+/g, '\r\n')
+    } catch (error) {}
+  }
+
+  async function fillClientData(clientId) {
+    if (!clientId) return
+
+    try {
+      const res = await getRequest({
+        extension: SaleRepository.Client.get,
+        parameters: `_recordId=${clientId}`
+      })
+      formik.setFieldValue('currencyId', res?.record?.currencyId)
+      formik.setFieldValue('spId', res?.record?.spId)
+      formik.setFieldValue('ptId', res?.record?.ptId)
+      formik.setFieldValue('plId', res?.record?.plId)
+      formik.setFieldValue('szId', res?.record?.szId)
+      formik.setFieldValue('shipToAddressId', res?.record?.shipAddressId)
+      formik.setFieldValue('billToAddressId', res?.record?.billAddressId)
+      const shipAdd = await getAddress(res?.record?.shipAddressId)
+      const billAdd = await getAddress(res?.record?.billAddressId)
+      formik.setFieldValue('shipAddress', shipAdd)
+      formik.setFieldValue('billAddress', billAdd)
```

Comment:
try using:
 formik.setValues({
    ...formik.values,
    shipAddress: await getAddress(record?.shipAddressId),
    billAddress: await getAddress(record?.billAddressId)
    ........
  });

Reply:
why should I retrieve all formik values to update only some fields? I guess it's better to update specific fields only